api/server/cluster.go

```diff
@@ -20,10 +21,21 @@ const (
 
 type clusterApi struct {
 	restBase
+	SecretManager *secrets.Manager
 }
 
-func newClusterAPI() restServer {
-	return &clusterApi{restBase{version: cluster.APIVersion, name: "Cluster API"}}
+type ClusterServerConfiguration struct {
```

Comment:
Could you add comments to this to why it is used and to its variables so that the callers know how to use it?